substra/sdk/client.py

```diff
@@ -97,24 +95,34 @@ def __init__(self, config_path=None, profile_name=None, user_path=None,
         # set current logged user if exists
         self.set_user()
 
+        if token:
+            self.set_token(token)
+
     @logit
-    def login(self):
+    def login(self, username, password):
         """Login.
 
         Allow to login to a remote server.
 
-        After setting your configuration with `substra config` using `-u` and `-p`
-        Launch `substra login`
-        You will get a token which will be stored by default in `~/.substra-user`
+        After setting your configuration with `substra config`, launch `substra login`.
+        You will be prompted for your username and password and get a token which will be
+        stored by default in `~/.substra-user`
         You can change that thanks to the --user option (works like the --profile option)
 
         """
 
         if not self._current_profile:
             raise exceptions.SDKException("No profile defined")
 
-        res = self.client.login()
+        res = self.client.login(username, password)
         token = res.json()['token']
+        self.set_token(token)
+        return token
+
+    def set_token(self, token):
```

Comment:
This method could be private as it's not needed to the end user. 

Having smaller public interfaces is easier to maintain (we can do more changes without breaking the current interface).

Reply:
Sure!